src/celengine/stardb.cpp

```diff
@@ -666,14 +704,13 @@ bool StarDatabaseBuilder::loadBinary(std::istream& in)
         if (std::string_view(header.data() + offsetof(StarsDatHeader, magic), STARSDAT_MAGIC.size()) != STARSDAT_MAGIC) { return false; }
 
         // Verify the version
-        std::uint16_t version;
-        std::memcpy(&version, header.data() + offsetof(StarsDatHeader, version), sizeof(version));
-        LE_TO_CPU_INT16(version, version);
-        if (version != 0x0100) { return false; }
+
+        if (auto version = readIntLE<std::uint16_t>(header.data() + offsetof(StarsDatHeader, version));
+            version != 0x0100)
```

Comment:
it seems we should add a constant for DB versions instead of magic number here and there

Reply:
also add {} here for readability